### ISSUE TITLE
Slow loading of pages with larger data

### ISSUE THREAD

**Issue by jteresco** [TravelMapping/Web#104]
Pages like mapview.php, system.php, and region.php can load extremely slowly even for what I'd consider moderately large regions or systems.  This is a note to look into what is causing the slowness, and if it turns out to be the DB queries as I suspect, to determine if more data should be computed at site update time and stored in the DB, or if the DB structure and/or queries need improvement, or if it's just slow because it's a lot of data to deal with.

**Comment by michihdeu** [comment]
I didn't wanna complain but that's really annoying. On loading and unloading my web broswer (FF) is usually blocked for some seconds...

I often load maps with more than one region, e.g. http://tm.teresco.org/user/mapview.php?u=michih&rg=AUT,CHE,DEU,FRA,ITA. This map was used for plotting cheh system to see concurrent routes or identical waypoints of neighboring systems. In addition, I use it for planning travels.

**Comment by jteresco** [comment]
I feel like something changed to slow things.  I used to be able to load several good-sized regions and still have to wait just a few seconds.  Now, I can't get a map with DEU only to load.  I will look into it when I can but that's probably not for a few weeks.

**Comment by michihdeu** [comment]
I don't think that it's notable slower. Of course, DEU region is slower because deub system is in preview now and yes, it's sometimes not loaded at all but I had the same behavior from the beginning... I was used to load this map and it usually worked. I tried it right now, and it worked:

http://tm.teresco.org/user/mapview.php?rg=AUT,BEL,CHE,CZE,DEU,DNK,ESP,FRA,HUN,ITA,LUX,NOR,NLD,POL,ROU,SVN,SWE&u=michih

It's OK with me but I think other (new) users might be discouraged to submit files...

**Comment by Thing342** [comment]
Pretty sure it's always been fairly slow, but I've been thinking about how to return the basic page first and then inject the waypoints from a separate request. Not totally sure how (or when i'll find the time) to do this, however.

**Comment by jteresco** [comment]
It was definitely always a little to very slow depending on the amount of data loading.  It's a bit odd, however, that this morning my attempts to see DEU and AUT loaded up within a few seconds.  Last night, I was waiting several minutes and got no indication of progress.

**Comment by yakra** [comment]
I noticed slowness when the most recent batch of the changes to the HB went live. At first I thought it may have been some changes on the site, but even using WPTedit now, is very slow to respond. I think some changes have been made to the underlying Google Maps API. :(
Even the HB is practically unusable for me now.

Well. Looks like it's time to upgrade my computer, then.

**Comment by jteresco** [comment]
I'm suspecting things might be changing behind the scenes with the API, since I'm able to bring up MUCH larger data today in mapview.  I'm not saying it's blazing fast but it seems much more usable.

**Comment by michihdeu** [comment]
I've reviewed my cheh routes this morning with the CHM editor. Sometimes I had to wait more than 60 seconds till the map was loaded. It was often just (partially) grey after jumping from one waypoint to the next one. The CHM graphs had no problem and were always loaded immediately. I wanna give up but suddenly the maps were loaded very quick again.

In addition, there's the memory leak which happens in FF and IE. I've opened a TM map in FF without touching it. It's just because I wanna check concurrent E roads if necessary. Memory usage grew from 200MB to 1GB after 3-4 hours. The memory usage in IE is less although I've loaded a lot of data in CHM editor. I close IE when memory usage is ~600MB (after 2-3 hours).

I've worked with wikimapia.org for years. Sometimes maps are loaded damn slow. The same happens to TM.
